WordPress/src/main/java/org/wordpress/android/ui/photopicker/PhotoPickerFragment.java

```diff
@@ -74,49 +85,48 @@
         Bundle args = new Bundle();
+        args.putBoolean(ARG_ALLOW_MULTI_SELECT, options.contains(PhotoPickerOption.ALLOW_MULTI_SELECT));
+        args.putBoolean(ARG_PHOTOS_ONLY, options.contains(PhotoPickerOption.PHOTOS_ONLY));
+        args.putBoolean(ARG_DEVICE_ONLY, options.contains(PhotoPickerOption.DEVICE_ONLY));
+
         PhotoPickerFragment fragment = new PhotoPickerFragment();
         fragment.setPhotoPickerListener(listener);
-        if (options != null) {
-            args.putBoolean(ARG_ALLOW_MULTI_SELECT, options.contains(PhotoPickerOption.ALLOW_MULTI_SELECT));
-            args.putBoolean(ARG_PHOTOS_ONLY, options.contains(PhotoPickerOption.PHOTOS_ONLY));
-            args.putBoolean(ARG_DEVICE_ONLY, options.contains(PhotoPickerOption.DEVICE_ONLY));
-        }
         fragment.setArguments(args);
         return fragment;
     }
 
     @Override
     public void onCreate(@Nullable Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
-        if (savedInstanceState != null) {
-            mAllowMultiSelect = savedInstanceState.getBoolean(ARG_ALLOW_MULTI_SELECT, false);
-            mPhotosOnly = savedInstanceState.getBoolean(ARG_PHOTOS_ONLY, false);
-            mDeviceOnly = savedInstanceState.getBoolean(ARG_DEVICE_ONLY, false);
+        Bundle args = getArguments();
+        if (args != null) {
+            mAllowMultiSelect = args.getBoolean(ARG_ALLOW_MULTI_SELECT, false);
+            mPhotosOnly = args.getBoolean(ARG_PHOTOS_ONLY, false);
+            mDeviceOnly = args.getBoolean(ARG_DEVICE_ONLY, false);
         }
     }
 
     @Override
     public void setArguments(Bundle args) {
         super.setArguments(args);
-        mAllowMultiSelect = args != null && args.getBoolean(ARG_ALLOW_MULTI_SELECT);
-        mPhotosOnly = args != null && args.getBoolean(ARG_PHOTOS_ONLY);
-        mDeviceOnly = args != null && args.getBoolean(ARG_DEVICE_ONLY);
-    }
-
-    @Override
-    public void onSaveInstanceState(Bundle outState) {
-        outState.putBoolean(ARG_ALLOW_MULTI_SELECT, mAllowMultiSelect);
-        outState.putBoolean(ARG_PHOTOS_ONLY, mPhotosOnly);
-        outState.putBoolean(ARG_DEVICE_ONLY, mDeviceOnly);
-        super.onSaveInstanceState(outState);
+        mAllowMultiSelect = args.getBoolean(ARG_ALLOW_MULTI_SELECT);
+        mPhotosOnly = args.getBoolean(ARG_PHOTOS_ONLY);
+        mDeviceOnly = args.getBoolean(ARG_DEVICE_ONLY);
     }
 
-    public void setOptions(EnumSet<PhotoPickerOption> options) {
-        mAllowMultiSelect = options != null && options.contains(PhotoPickerOption.ALLOW_MULTI_SELECT);
-        mPhotosOnly = options != null && options.contains(PhotoPickerOption.PHOTOS_ONLY);
+    public void setOptions(@NonNull EnumSet<PhotoPickerOption> options) {
+        mAllowMultiSelect = options.contains(PhotoPickerOption.ALLOW_MULTI_SELECT);
+        mPhotosOnly = options.contains(PhotoPickerOption.PHOTOS_ONLY);
```

Comment:
I might be wrong, but it seems that `mDeviceOnly` is missing here. 

Minor: Wondering why there is this method to refresh/set Fragment parameters. Is there a way to switch the picker mode after its creation? I see it's used by the main activity in `onCreate`, when the fragment is already available.